meraki_v0/rest_session.py

```diff
@@ -50,11 +57,53 @@ def __init__(
         elif self._base_url[-1] == '/':
             self._base_url = self._base_url[:-1]
 
+        # Generate extended portion of the User-Agent
+        user_agent_extended = be_geo_id
+        user_agent_extended = {}
+
+        # Mimic pip system data collection per https://github.com/pypa/pip/blob/master/src/pip/_internal/network/session.py
+        user_agent_extended['implementation'] = {
+                "name": platform.python_implementation(),
+            }
+
+        if user_agent_extended["implementation"]["name"] == 'CPython':
```

Comment:
This whole block can be shortened

```python
        if user_agent_extended["implementation"]["name"] in ('CPython','Jython','IronPython'):
            user_agent_extended["implementation"]["version"] = platform.python_version()
        elif user_agent_extended["implementation"]["name"] == 'PyPy':
            if sys.pypy_version_info.releaselevel == 'final':
                pypy_version_info = sys.pypy_version_info[:3]
            else:
                pypy_version_info = sys.pypy_version_info
            user_agent_extended["implementation"]["version"] = ".".join(
                [str(x) for x in pypy_version_info]
            )
```

Reply:
Good catch!  The original was literally copied/pasted from the pip client User-Agent processing in an attempt to maintain some consistency with that.  If we think that having more streamlined code is more important than that consistency, I'm fine changing it to the suggested block, but I wanted to provide some background as to why it is the way it is.

Thoughts?  Consistency or streamlined?

Reply:
I can't speak for pip, but my guess would be, that it was just a simple copy & paste or that the "platform.python_version()" methods were different in the past for the implementations.

Also would you mind to extract this user-agent part as a function? That way we could use it for the async implementation too without copy & paste =)